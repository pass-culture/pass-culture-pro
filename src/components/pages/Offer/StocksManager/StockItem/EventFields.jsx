import moment from 'moment'
import PropTypes from 'prop-types'
import React, { Component, Fragment } from 'react'

import { DateField, HiddenField, TimeField } from 'components/layout/form'

export class EventFields extends Component {
  /*
  componentDidUpdate(prevProps) {
    this.handleCrossingEndDatetime()
  }

  handleCrossingEndDatetime = () => {
    const {
      dispatch,
      formBeginningDatetime,
      formEndDatetime,
      stockFormKey,
    } = this.props

    if (formEndDatetime >= formBeginningDatetime) {
      return
    }

    const endDatetime = getDatetimeOneDayAfter(formBeginningDatetime)

    dispatch(mergeForm(`stock${stockFormKey}`, { endDatetime }))
  }
  */

  render() {
    const { beginningMinDate, readOnly, stocks, tz } = this.props
    const highlightDates = (stocks || []).map(stock => stock.beginningDatetime)

    return (
      <Fragment>
        <td>
          <DateField
            highlightDates={highlightDates}
            minDate={beginningMinDate || moment()}
            name="beginningDatetime"
            readOnly={readOnly}
            required
            title="Date"
          />
        </td>
        <td>
          <TimeField
            name="beginningTime"
            patchKey="beginningDatetime"
            readOnly={readOnly}
            required
            title="Heure"
            tz={tz}
          />
        </td>
        <td>
          <HiddenField name="endDatetime" />
          <TimeField
            name="endTime"
            patchKey="endDatetime"
            readOnly={readOnly}
            required
            title="Heure de fin"
            tz={tz}
          />
        </td>
      </Fragment>
    )
  }
}

EventFields.defaultProps = {
  beginningMinDate: null,
  stocks: null,
}

EventFields.propTypes = {
  beginningMinDate: PropTypes.string,
  dispatch: PropTypes.func.isRequired,
  stocks: PropTypes.array,
}

export default EventFields
