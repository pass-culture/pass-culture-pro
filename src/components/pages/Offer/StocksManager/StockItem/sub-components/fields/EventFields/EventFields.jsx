import moment from 'moment'
import PropTypes from 'prop-types'
import React, { Fragment } from 'react'

import DateField from 'components/layout/form/fields/DateField/DateField'
import TimeField from 'components/layout/form/fields/TimeField'

const EventFields = ({ beginningMinDate, readOnly, stocks, timezone }) => {
  const highlightDates = (stocks || []).map(stock => stock.beginningDatetime)

  return (
    <Fragment>
      <td>
        <DateField
          highlightDates={highlightDates}
          minDate={beginningMinDate || moment()}
          name="beginningDatetime"
          readOnly={readOnly}
          required
          timezone={timezone}
          title="Date"
        />
      </td>
      <td>
        <TimeField
          name="beginningTime"
          patchKey="beginningDatetime"
          readOnly={readOnly}
          required
          title="Heure"
        />
      </td>
    </Fragment>
  )
}

EventFields.defaultProps = {
  beginningMinDate: null,
  readOnly: true,
  stocks: null,
  timezone: null,
}

EventFields.propTypes = {
  beginningMinDate: PropTypes.string,
  readOnly: PropTypes.bool,
  stocks: PropTypes.arrayOf(PropTypes.shape()),
  timezone: PropTypes.string,
}

export default EventFields
