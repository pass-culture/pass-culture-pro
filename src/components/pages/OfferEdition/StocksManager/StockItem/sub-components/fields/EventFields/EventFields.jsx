import moment from 'moment'
import PropTypes from 'prop-types'
import React, { Fragment } from 'react'

import DateField from '../../../../../../../layout/form/fields/DateField'
import HiddenField from '../../../../../../../layout/form/fields/HiddenField'
import TimeField from '../../../../../../../layout/form/fields/TimeField'

const EventFields = ({ beginningMinDate, readOnly, stocks, timezone }) => {
  const highlightDates = (stocks || []).map(stock => stock.beginningDatetime)

  return (
    <Fragment>
      <td>
        <DateField
          highlightDates={highlightDates}
          minDate={beginningMinDate || moment()}
          name="beginningDatetime"
          readOnly={readOnly}
          required
          timezone={timezone}
          title="Date"
        />
      </td>
      <td>
        <TimeField
          name="beginningTime"
          patchKey="beginningDatetime"
          readOnly={readOnly}
          required
          title="Heure"
        />
      </td>
      <td>
        <HiddenField name="endDatetime" />
        <TimeField
          name="endTime"
          patchKey="endDatetime"
          readOnly={readOnly}
          required
          title="Heure de fin"
        />
      </td>
    </Fragment>
  )
}

EventFields.defaultProps = {
  beginningMinDate: null,
  readOnly: true,
  stocks: null,
  timezone: null,
}

EventFields.propTypes = {
  beginningMinDate: PropTypes.string,
  readOnly: PropTypes.bool,
  stocks: PropTypes.arrayOf(PropTypes.shape()),
  timezone: PropTypes.string,
}

export default EventFields
