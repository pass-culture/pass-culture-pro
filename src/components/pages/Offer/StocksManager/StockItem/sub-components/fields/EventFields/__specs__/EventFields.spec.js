import { shallow } from 'enzyme'
import React from 'react'

import EventFields from '../EventFields'

describe('src | components | pages | Offer | StockItem | EventFields', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const initialProps = {
        beginningMinDate: '2019-03-29T01:56:55.610Z',
        dispatch: jest.fn(),
        parseFormChild: jest.fn(),
      }

      // when
      const wrapper = shallow(<EventFields {...initialProps} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
