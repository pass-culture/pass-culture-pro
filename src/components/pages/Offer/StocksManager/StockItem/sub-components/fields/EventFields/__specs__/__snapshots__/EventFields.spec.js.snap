// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Offer | StockItem | EventFields should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventFields
    beginningMinDate="2019-03-29T01:56:55.610Z"
    dispatch={[MockFunction]}
    parseFormChild={[MockFunction]}
    readOnly={true}
    stocks={null}
    timezone={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <td>
          <DateField
            autoComplete={false}
            className=""
            clearable={false}
            dateFormat="DD/MM/YYYY"
            datePickerClassName="date"
            disabled={false}
            highlightDates={Array []}
            id={null}
            label=""
            locale="fr"
            minDate="2019-03-29T01:56:55.610Z"
            name="beginningDatetime"
            placeholder="Please enter a value"
            readOnly={true}
            renderValue={[Function]}
            required={true}
            timezone={null}
            title="Date"
            validate={null}
          />
        </td>,
        <td>
          <TimeField
            autoComplete={false}
            className=""
            disabled={false}
            id={null}
            label=""
            locale="fr"
            name="beginningTime"
            patchKey="beginningDatetime"
            placeholder="Please enter a value"
            readOnly={true}
            renderValue={[Function]}
            required={true}
            title="Heure"
            validate={null}
          />
        </td>,
        <td>
          <HiddenField
            name="endDatetime"
            validator={[Function]}
          />
          <TimeField
            autoComplete={false}
            className=""
            disabled={false}
            id={null}
            label=""
            locale="fr"
            name="endTime"
            patchKey="endDatetime"
            placeholder="Please enter a value"
            readOnly={true}
            renderValue={[Function]}
            required={true}
            title="Heure de fin"
            validate={null}
          />
        </td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <DateField
            autoComplete={false}
            className=""
            clearable={false}
            dateFormat="DD/MM/YYYY"
            datePickerClassName="date"
            disabled={false}
            highlightDates={Array []}
            id={null}
            label=""
            locale="fr"
            minDate="2019-03-29T01:56:55.610Z"
            name="beginningDatetime"
            placeholder="Please enter a value"
            readOnly={true}
            renderValue={[Function]}
            required={true}
            timezone={null}
            title="Date"
            validate={null}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoComplete": false,
            "className": "",
            "clearable": false,
            "dateFormat": "DD/MM/YYYY",
            "datePickerClassName": "date",
            "disabled": false,
            "highlightDates": Array [],
            "id": null,
            "label": "",
            "locale": "fr",
            "minDate": "2019-03-29T01:56:55.610Z",
            "name": "beginningDatetime",
            "placeholder": "Please enter a value",
            "readOnly": true,
            "renderValue": [Function],
            "required": true,
            "timezone": null,
            "title": "Date",
            "validate": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <TimeField
            autoComplete={false}
            className=""
            disabled={false}
            id={null}
            label=""
            locale="fr"
            name="beginningTime"
            patchKey="beginningDatetime"
            placeholder="Please enter a value"
            readOnly={true}
            renderValue={[Function]}
            required={true}
            title="Heure"
            validate={null}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoComplete": false,
            "className": "",
            "disabled": false,
            "id": null,
            "label": "",
            "locale": "fr",
            "name": "beginningTime",
            "patchKey": "beginningDatetime",
            "placeholder": "Please enter a value",
            "readOnly": true,
            "renderValue": [Function],
            "required": true,
            "title": "Heure",
            "validate": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <HiddenField
              name="endDatetime"
              validator={[Function]}
            />,
            <TimeField
              autoComplete={false}
              className=""
              disabled={false}
              id={null}
              label=""
              locale="fr"
              name="endTime"
              patchKey="endDatetime"
              placeholder="Please enter a value"
              readOnly={true}
              renderValue={[Function]}
              required={true}
              title="Heure de fin"
              validate={null}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "endDatetime",
              "validator": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoComplete": false,
              "className": "",
              "disabled": false,
              "id": null,
              "label": "",
              "locale": "fr",
              "name": "endTime",
              "patchKey": "endDatetime",
              "placeholder": "Please enter a value",
              "readOnly": true,
              "renderValue": [Function],
              "required": true,
              "title": "Heure de fin",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "td",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <td>
            <DateField
              autoComplete={false}
              className=""
              clearable={false}
              dateFormat="DD/MM/YYYY"
              datePickerClassName="date"
              disabled={false}
              highlightDates={Array []}
              id={null}
              label=""
              locale="fr"
              minDate="2019-03-29T01:56:55.610Z"
              name="beginningDatetime"
              placeholder="Please enter a value"
              readOnly={true}
              renderValue={[Function]}
              required={true}
              timezone={null}
              title="Date"
              validate={null}
            />
          </td>,
          <td>
            <TimeField
              autoComplete={false}
              className=""
              disabled={false}
              id={null}
              label=""
              locale="fr"
              name="beginningTime"
              patchKey="beginningDatetime"
              placeholder="Please enter a value"
              readOnly={true}
              renderValue={[Function]}
              required={true}
              title="Heure"
              validate={null}
            />
          </td>,
          <td>
            <HiddenField
              name="endDatetime"
              validator={[Function]}
            />
            <TimeField
              autoComplete={false}
              className=""
              disabled={false}
              id={null}
              label=""
              locale="fr"
              name="endTime"
              patchKey="endDatetime"
              placeholder="Please enter a value"
              readOnly={true}
              renderValue={[Function]}
              required={true}
              title="Heure de fin"
              validate={null}
            />
          </td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <DateField
              autoComplete={false}
              className=""
              clearable={false}
              dateFormat="DD/MM/YYYY"
              datePickerClassName="date"
              disabled={false}
              highlightDates={Array []}
              id={null}
              label=""
              locale="fr"
              minDate="2019-03-29T01:56:55.610Z"
              name="beginningDatetime"
              placeholder="Please enter a value"
              readOnly={true}
              renderValue={[Function]}
              required={true}
              timezone={null}
              title="Date"
              validate={null}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoComplete": false,
              "className": "",
              "clearable": false,
              "dateFormat": "DD/MM/YYYY",
              "datePickerClassName": "date",
              "disabled": false,
              "highlightDates": Array [],
              "id": null,
              "label": "",
              "locale": "fr",
              "minDate": "2019-03-29T01:56:55.610Z",
              "name": "beginningDatetime",
              "placeholder": "Please enter a value",
              "readOnly": true,
              "renderValue": [Function],
              "required": true,
              "timezone": null,
              "title": "Date",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TimeField
              autoComplete={false}
              className=""
              disabled={false}
              id={null}
              label=""
              locale="fr"
              name="beginningTime"
              patchKey="beginningDatetime"
              placeholder="Please enter a value"
              readOnly={true}
              renderValue={[Function]}
              required={true}
              title="Heure"
              validate={null}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoComplete": false,
              "className": "",
              "disabled": false,
              "id": null,
              "label": "",
              "locale": "fr",
              "name": "beginningTime",
              "patchKey": "beginningDatetime",
              "placeholder": "Please enter a value",
              "readOnly": true,
              "renderValue": [Function],
              "required": true,
              "title": "Heure",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <HiddenField
                name="endDatetime"
                validator={[Function]}
              />,
              <TimeField
                autoComplete={false}
                className=""
                disabled={false}
                id={null}
                label=""
                locale="fr"
                name="endTime"
                patchKey="endDatetime"
                placeholder="Please enter a value"
                readOnly={true}
                renderValue={[Function]}
                required={true}
                title="Heure de fin"
                validate={null}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "endDatetime",
                "validator": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoComplete": false,
                "className": "",
                "disabled": false,
                "id": null,
                "label": "",
                "locale": "fr",
                "name": "endTime",
                "patchKey": "endDatetime",
                "placeholder": "Please enter a value",
                "readOnly": true,
                "renderValue": [Function],
                "required": true,
                "title": "Heure de fin",
                "validate": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "td",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
